4788/src/main/cpp/Shooter.cpp

```diff
@@ -1,82 +1,98 @@
+  setFlyWheelPower = power;
+}
 
-  switch (_teleopShooter) {
-    case TeleopShooter::kAuto:
-      //left bumper for close shot, right bumper for far shot, POV button 
-      // if (_contGroup.Get(ControlMap::shortShoot)) {
-      //   speed(8, dt);
-      // }
-      break;
-    case TeleopShooter::kIdle:
-
-      break;
-    case TeleopShooter::kManual:
-
-      manualControl(dt);
-
-      break;
-    case TeleopShooter::kTesting:
-      testing(dt);
-      break;
-    default:
-      break;
-  }
+void Shooter::setIndex(double power) {
+  setIndexPower = power;
+}
+
+void Shooter::setAuto(double goal, double dt) {
+  pidShooterSpeed = speed(flyWheelGoal, dt);
+  _teleopShooter = TeleopShooter::kAuto;
+  flyWheelGoal = goal;
+  //call the PID loop
 }
 
-//TODO @Anna figure out PID algorithm stuff
-/**
-  * Needs to be a closed loop PID system 
-  * 
-  */
-double Shooter::speed(double metersPerSecond, double dt) {
+void Shooter::updateShooter(double dt) {
 
-  // double input = _shooterSystem._flyWheel.encoder->GetAngularVelocity();
+  switch (_teleopShooter) {
+  case TeleopShooter::kManual:
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(setFlyWheelPower * 12);
 
-  // ControlMap::error = ControlMap::goal - input;
-  // ControlMap::derror = (ControlMap::error - ControlMap::previousError) / dt;
-  // ControlMap::sum = ControlMap::sum + ControlMap::error * dt;
+    _shooterSystem.indexWheel.Set(setIndexPower);
+    std::cout << "index speed" << setIndexPower << std::endl;
 
-  // ControlMap::ouput = ControlMap::kp * ControlMap::error + ControlMap::ki * ControlMap::sum + ControlMap::kd * ControlMap::derror;
-  // ControlMap::previousError = ControlMap::error;
 
-  // return ControlMap::output;
-}
+    break;
+
+  case TeleopShooter::kIdle:
+
+    _shooterSystem.indexWheel.Set(setIndexPower);
 
 
-/**
-  * Left trigger controls the shooter manually
-  */
-void Shooter::manualControl(double dt) {
-  shooterManualSpeed = fabs(_contGroup.Get(ControlMap::shooterManualSpin)) > ControlMap::triggerDeadzone ? _contGroup.Get(ControlMap::shooterManualSpin) : 0;
+    break;
 
-  _shooterSystem.shooterGearbox.transmission->SetVoltage(shooterManualSpeed);
+  case TeleopShooter::kTesting:
+
+  
+
+    break;
+
+  case TeleopShooter::kAuto:
+    _shooterSystem.indexWheel.Set(setIndexPower);
+
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(pidShooterSpeed);
+    std::cout << "shooter speed" << _shooterSystem.rightFlyWheelMotor.GetEncoderAngularVelocity() << std::endl; 
+    break;
+
+  default:
+    break;
+  }
+}
+
+void Shooter::update(double dt) {
+  updateShooter(dt);
 }
 
-/**
-  * for testing the shooter
-  */
-void Shooter::testing(double dt) {
 
-  shooterTestingSpeed = fabs(_contGroup.Get(ControlMap::shooterManualSpin)) > ControlMap::triggerDeadzone ? _contGroup.Get(ControlMap::shooterManualSpin) : 0;
+double sum = 0;
+double previousError = 0;
+double previousOutput = 0;
+double Shooter::speed(double goal, double dt) {
```

Comment:
Rename to `calculate` or something similar - `speed` on its own is a noun, not a verb and makes no sense in this context (particularly because speed is an internal variable that is being measured and has no relation to the parameters and returns of this function).